### PR TITLE
docs, multi-net policy: provide user documentation

### PR DESCRIPTION
**Pull request by maiqueb**
**- What this PR does and why is it needed**

This PR introduces minimal user documentation for multi-network policies.

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Review by npinaeva — COMMENTED** [review]
I finally understood how this mysterious multi-networkpolicy looks like, thanks @maiqueb!

File: docs/multi-homing.md [hunk 1]
```diff
@@ -234,6 +234,64 @@ spec:
...
   attachment configuration does **not** feature subnets.
 
+## Multi-network Policies
+OVN-Kubernetes implements native support for
+[multi-networkpolicy](https://github.com/k8snetworkplumbingwg/multi-networkpolicy),
+an API providing
+[network policy](https://kubernetes.io/docs/concepts/services-networking/network-policies/)
+features for secondary networks.
+
+To configure traffic segregation for your pods, the user must:
```
[the remark this answers is not in the document]
Comment (maiqueb) [reply docs/multi-homing.md:244]: Thank you for setting my nomenclature straight.

File: docs/multi-homing.md [hunk 2]
```diff
@@ -234,6 +234,64 @@ spec:
...
+          role: trusted # only pods on namespaces with this label will be allowed on port 9000
+  policyTypes:
+    - Ingress
+```
+
+Please note the `MultiNetworkPolicy` has the **exact same** API of the native
+`networking.k8s.io/v1` `NetworkPolicy`object; check its documentation for more
+information.
+
+**Note:** It currently **requires** the secondary networks defined to have a defined
```
[the remark this answers is not in the document]
Comment (maiqueb) [reply docs/multi-homing.md:292]: Yes, that's exactly what I meant. 

I'll rephrase to make it clearer.

File: docs/multi-homing.md [hunk 3]
```diff
@@ -234,6 +234,64 @@ spec:
...
+        "topology": "layer2",
+        "type": "ovn-k8s-cni-overlay",
+        "subnets": "192.168.100.0/24"
+    }'
+---
+apiVersion: k8s.cni.cncf.io/v1beta1
+kind: MultiNetworkPolicy
+metadata:
+  annotations:
+    k8s.v1.cni.cncf.io/policy-for: tenant-blue
```
Comment (maiqueb) [inline docs/multi-homing.md:270]: hm, let's add a comment here as well.

File: docs/multi-homing.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (maiqueb) [reply docs/multi-homing.md]: Done.

File: docs/multi-homing.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (maiqueb) [reply docs/multi-homing.md]: Done.

File: docs/multi-homing.md (hunk 3 above)
Comment (maiqueb) [reply docs/multi-homing.md]: Done.

File: docs/multi-homing.md [hunk 4]
```diff
@@ -234,6 +234,67 @@ spec:
...
+  policyTypes:
+    - Ingress
+```
+
+Please note the `MultiNetworkPolicy` has the **exact same** API of the native
+`networking.k8s.io/v1` `NetworkPolicy`object; check its documentation for more
+information.
+
+**Note:** It is currently **required** for the `net-attach-def`s referred to by
+the `k8s.v1.cni.cncf.io/policy-for` annotation to have the `subnet` attribute
```
Comment (npinaeva) [inline docs/multi-homing.md:295]: is it `subnet` or `subnets`?
also, let's say something like "`subnets` attribute in its `spec.config` defined". It's probably obvious for more experienced multi-network users where to look for "subnet", but I would question myself if I am looking at the right field :)

Comment (maiqueb) [reply docs/multi-homing.md]: Done.

File: docs/multi-homing.md [hunk 5]
```diff
@@ -234,6 +234,67 @@ spec:
...
+        "topology": "layer2",
+        "type": "ovn-k8s-cni-overlay",
+        "subnets": "192.168.100.0/24"
+    }'
+---
+apiVersion: k8s.cni.cncf.io/v1beta1
+kind: MultiNetworkPolicy
+metadata:
+  annotations:
+    k8s.v1.cni.cncf.io/policy-for: tenant-blue # indicates the net-attach-defs this policy applies to
```
Comment (cathy-zhou) [inline docs/multi-homing.md:272]: should be default/tenant-blue in your case?

File: docs/multi-homing.md [hunk 6]
```diff
@@ -234,6 +234,67 @@ spec:
...
+[network policy](https://kubernetes.io/docs/concepts/services-networking/network-policies/)
+features for secondary networks.
+
+To configure pod isolation, the user must:
+- provision a `network-attachment-definition`.
+- provision a `MultiNetworkPolicy` indicating to which secondary networks it
+  applies via the
+  [policy-for](https://github.com/k8snetworkplumbingwg/multi-networkpolicy#policy-for-annotation)
+  annotation.
+
```
Comment (cathy-zhou) [inline docs/multi-homing.md:250]: needs to set OVN_MULTI_NETWORK_ENABLE to true. Also, after https://github.com/ovn-org/ovn-kubernetes/pull/3555, needs to set new flag to true.

**Review by npinaeva — DISMISSED** [review]
looks good from my side

File: docs/multi-homing.md (hunk 5 above)
Comment (maiqueb) [reply docs/multi-homing.md]: Good point; done.

File: docs/multi-homing.md (hunk 6 above)
Comment (maiqueb) [reply docs/multi-homing.md:250]: Done.

**Review by npinaeva — APPROVED** [review]
still lgtm :)

**Review by cathy-zhou — APPROVED** [review]

**Pull request merged**